File: kar-outdoor-ui/src/app/common/purchase.ts
import {Customer} from './customer';
import {Address} from './address';
import {OrderItem} from './order-item';
import {Order} from './order';
import {CreditCard} from './credit-card';

export class Purchase {

  constructor(public customer: Customer,
              public shippingAddress: Address,
              public billingAddress: Address,
              public order: Order,
              public creditCard : CreditCard,
              public orderItems: OrderItem[],
              public totalQuantity: number,
              public totalPrice: number,) {

  }
}

File: src/main/java/com/karoutdoor/dto/Purchase.java
package com.karoutdoor.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Purchase {

    private CustomerDto customer;
    private AddressDto shippingAddress;
    private AddressDto billingAddress;

    private List<OrderItemDto> orderItems;
    private CreditCard creditCard;

    private BigDecimal totalPrice;
    private Integer totalQuantity;
}
